src/foam/nanos/crunch/connection/ConnectedCapabilityDAO.js

```diff
@@ -39,6 +43,36 @@ foam.CLASS({
       name: 'put_',
       javaCode: `
         // TODO: add logic to disprese FObjectArray to lower level capabilities
+        DAO capabilityDAO = (DAO) x.get("capabilityDAO");
+        ConnectedCapability sentCap = (ConnectedCapability) obj;
+        FlatCapability flatCap = (FlatCapability)
+          capabilityDAO.find(sentCap.getFlatCapabilityId());
```

Comment:
sentCap is the ConnectedCapability object. It looks like it doesn't implement that method - should it?

Reply:
If you look at the generated code, you can see this method:

`  public foam.nanos.crunch.connection.FlatCapability findFlatCapabilityId(foam.core.X x) {
    return (foam.nanos.crunch.connection.FlatCapability)((foam.dao.DAO) x.get("flatCapabilityDAO")).find_(x, (Object) getFlatCapabilityId());
  }`

Maybe the field should be named flatCapability instead of flatCapabilityId so this method makes sense.